sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -517,12 +456,19 @@ def dead_letter(self, description=None):
         :raises: ~azure.servicebus.common.errors.SessionLockExpired if session lock has already expired.
         :raises: ~azure.servicebus.common.errors.MessageSettleFailed if message settle operation fails.
         """
+        # pylint: disable=protected-access
         self._is_live('dead-letter')
         details = {
             'deadletter-reason': str(description) if description else "",
```

Comment:
Is there a reason to populate both if it's.. the same thing?  (This begs the question what the intended semantic delta between those two inputs is)

Reply:
as per discussion, according to SB sepc, we should support both reason and description